File: Micro-Service/Src/Services/Shipping/Shipping.Domain/Customer.cs
namespace Shipping.Domain;
public sealed class Customer
{
    public Guid Id { get; private set; } = Guid.NewGuid();
    public string Name { get; private set; } = default!;
    public string Email { get; private set; } = default!;
    public string PhoneNumber { get; private set; } = default!;
    public string Address { get; private set; } = default!;

    private Customer() { }
    public Customer(string name, string email, string phoneNumber, string address)
    {
        Name = name;
        Email = email;
        PhoneNumber = phoneNumber;
        Address = address;
    }
}

File: MicroService-FrontEnd/Home/src/app/model/customer/customer.ts
interface CustomerReader
{
    id: string;
    name: string;
    address: string;
    phoneNumber: string;
    email: string;    
}

interface CustomerWriter
{
    name: string;
    address: string;
    phoneNumber: string;
    email: string;    
}

interface Customer
{
    id: string;
    name: string;
    address: string;
    phoneNumber: string;
    email: string;    
    gender: string;
    customerTypeId: number;
    hobbies: string[];
    countryCodes: string[];
}